services/provider/server/server.go

```diff
@@ -80,17 +82,32 @@ func NewOCSProviderServer(ctx context.Context, namespace string) (*OCSProviderSe
 		return nil, fmt.Errorf("failed to create new StorageClassRequest instance. %v", err)
 	}
 
+	version, err := semver.Make(ocsVersion.Version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse operator version as semver. %v", err)
+	}
+
 	return &OCSProviderServer{
 		client:                     client,
 		consumerManager:            consumerManager,
 		storageClassRequestManager: storageClassRequestManager,
 		namespace:                  namespace,
+		version:                    &version,
 	}, nil
 }
 
 // OnboardConsumer RPC call to onboard a new OCS consumer cluster.
 func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.OnboardConsumerRequest) (*pb.OnboardConsumerResponse, error) {
 
+	version, err := semver.Make(req.ClientOperatorVersion)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "malformed ClientPlatformVersion for client %q is provided. %v", req.ConsumerName, err)
+	}
+
+	if s.version.Minor != version.Minor || s.version.Major != version.Major {
+		return nil, status.Errorf(codes.InvalidArgument, "both major and minor version of server and client %q should match for onboarding process", req.ConsumerName)
+	}
```

Comment:
Is there a reason why we are not using `s.Compare`?

Reply:
- I don't remember it provides comparison for Major & Minor separately, will take a look.

Reply:
- changed to strict comparison which also checks patch versions and uses `Compare`
- as of now, I believe it's better to check patch versions as well due to the release cadence of provider mode feature
- let me know if we want to check only major & minor

Reply:
ok